src/vs/workbench/contrib/terminalContrib/suggest/browser/lspTerminalModelContentProvider.ts

```diff
@@ -0,0 +1,183 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Disposable, IDisposable, MutableDisposable } from '../../../../../base/common/lifecycle.js';
+import { IModelService } from '../../../../../editor/common/services/model.js';
+import { ILanguageService } from '../../../../../editor/common/languages/language.js';
+import { ITextModelContentProvider, ITextModelService } from '../../../../../editor/common/services/resolverService.js';
+import { URI } from '../../../../../base/common/uri.js';
+import { ITextModel } from '../../../../../editor/common/model.js';
+import { Schemas } from '../../../../../base/common/network.js';
+import { ICommandDetectionCapability, ITerminalCapabilityStore, TerminalCapability } from '../../../../../platform/terminal/common/capabilities/capabilities.js';
+import { ILspTerminalModelContentProvider } from '../../../../browser/lspTerminalCapability.js';
+import { IMarkerService } from '../../../../../platform/markers/common/markers.js';
+import { GeneralShellType, TerminalShellType } from '../../../../../platform/terminal/common/terminal.js';
+
+export const VSCODE_LSP_TERMINAL_PROMPT_TRACKER = 'vscode_lsp_terminal_prompt_tracker= {}\n';
+
+export class LspTerminalModelContentProvider extends Disposable implements ILspTerminalModelContentProvider, ITextModelContentProvider {
+	static readonly scheme = Schemas.vscodeTerminal;
+	private _commandDetection: ICommandDetectionCapability | undefined;
+	private _capabilitiesStore: ITerminalCapabilityStore;
+	private readonly _virtualTerminalDocumentUri: URI;
+	private readonly _markerFilterDisposable: IDisposable;
+	private _shellType: TerminalShellType | undefined;
+	private readonly _onCommandFinishedListener = this._register(new MutableDisposable());
+
+	constructor(
+		capabilityStore: ITerminalCapabilityStore,
+		terminalId: number,
+		virtualTerminalDocument: URI,
+		shellType: TerminalShellType | undefined,
+		@ITextModelService textModelService: ITextModelService,
+		@IModelService private readonly _modelService: IModelService,
+		@ILanguageService private readonly _languageService: ILanguageService,
+		@IMarkerService private readonly _markerService: IMarkerService,
+
+	) {
+		super();
+		this._register(textModelService.registerTextModelContentProvider(LspTerminalModelContentProvider.scheme, this));
+		this._capabilitiesStore = capabilityStore;
+		this._commandDetection = this._capabilitiesStore.get(TerminalCapability.CommandDetection);
+		this._registerTerminalCommandFinishedListener();
+		this._virtualTerminalDocumentUri = virtualTerminalDocument;
+
+		// Suppress diagnostics for REPL virtual document, similar to agentic mode.
+		this._markerFilterDisposable = this._markerService.installResourceFilter(
```

Comment:
I had thought this was no longer needed?

Reply:
Super correct, got rid of them here: [5d9f3b7](https://github.com/microsoft/vscode/pull/243528/commits/5d9f3b74854469938c2b74c2d62d7e8d74c71943)